CoreUtilities/src/au/gov/asd/tac/constellation/utilities/genericjsonio/JsonIO.java

```diff
@@ -211,6 +233,106 @@ public static void saveJsonPreferences(final Optional<String> saveDir, final Obj
         saveJsonPreferences(saveDir, Optional.empty(), rootNode, OBJECT_MAPPER);
     }
 
+    public static void saveJsonPreferencesWithKeyboardShortcut(final Optional<String> saveDir, final Object rootNode) {
+
+        ObjectMapper mapper = OBJECT_MAPPER;
+        final File preferenceDirectory = getPrefereceFileDirectory(saveDir);
+
+        // If the preference directory cannot be accessed then return
+        if (!preferenceDirectory.isDirectory()) {
+            NotifyDisplayer.display(
+                    String.format("Can't create preference directory '%s'.", preferenceDirectory),
+                    NotifyDescriptor.ERROR_MESSAGE
+            );
+
+            return;
+        }
+
+        //Record keyboard shortcut
+        Optional<String> ks = getDefaultKeyboardShortcut(preferenceDirectory);
+
+        // Ask the user to provide a file name        
+        Optional<String> userInputWithKs;
+
+        final Optional<KeyboardShortcutSelectionResult> ksResult = JsonIODialog.getPreferenceFileName(ks, preferenceDirectory);
+        if (ksResult.isPresent()) {
+            if (Objects.isNull(ksResult.get().getFileName())) {
+                return;
+            }
+
+            userInputWithKs = Optional.ofNullable(ksResult.get().getFileName());
+            ks = Optional.of("[" + ksResult.get().getKeyboardShortcut() + "]");
+
+        } else {
+            return;
+        }
+
+        final Optional<String> userInput = userInputWithKs;
```

Comment:
is there a reason why we need to create a new variable here instead of continuing to use `userInputWithKs`?

Reply:
no particular reason. probably assigned to get debugging and left out. fixed.